### PR TITLE
CI: Install more packages from conda-forge

### PR DESCRIPTION
**Pull request by jonashaag**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Let's see how this goes. Not sure if any of those pip installs are there on purpose or just because at the time of writing no conda-forge package was available.

### CODE REVIEW

**Comment by jonashaag** [comment]
Test failures seem to be unrelated?

File: ci/deps/azure-windows-39.yaml [hunk 1]
```diff
@@ -36,6 +36,5 @@ dependencies:
   - xlsxwriter
   - xlwt
   - pyreadstat
-  - pip
-  - pip:
-    - pyxlsb
+  - pyxlsb
+  - zstandard
```
Comment (jreback) [inline ci/deps/azure-windows-39.yaml:40]: we don't need this yet (open PR but will be added then)

File: ci/deps/circle-38-arm64.yaml [hunk 2]
```diff
@@ -15,7 +15,6 @@ dependencies:
   - numpy
   - python-dateutil
   - pytz
-  - pip
+  - zstandard
```
Comment (jreback) [inline ci/deps/circle-38-arm64.yaml:18]: same

File: requirements-dev.txt [hunk 3]
```diff
@@ -20,6 +20,11 @@ gitpython
 gitdb
 sphinx
 sphinx-panels
+numpydoc < 1.2
```
Comment (jreback) [inline requirements-dev.txt:23]: @pandas-dev/pandas-core do we have an issue about this? can we remove (independently than this PR)

**Review by jreback — CHANGES_REQUESTED** [review]
since CI is currently red will need a clean run

File: ci/deps/azure-windows-39.yaml (hunk 1 above)
Comment (jonashaag) [reply ci/deps/azure-windows-39.yaml:40]: oops

Comment (jonashaag) [inline ci/deps/azure-windows-39.yaml:40]:
```suggestion
```

File: ci/deps/circle-38-arm64.yaml (hunk 2 above)
Comment (jonashaag) [inline ci/deps/circle-38-arm64.yaml:18]:
```suggestion
```

File: requirements-dev.txt (hunk 3 above)
Comment (lithomas1) [reply requirements-dev.txt:23]: #39688

**Review by lithomas1 — APPROVED** [review]

File: ci/deps/azure-windows-39.yaml (hunk 1 above)
Comment (jreback) [reply ci/deps/azure-windows-39.yaml:40]: can you remove

File: ci/deps/circle-38-arm64.yaml (hunk 2 above)
Comment (jreback) [reply ci/deps/circle-38-arm64.yaml:18]: same

**Review by jreback — CHANGES_REQUESTED** [review]
@jonashaag can you make these changes and merge master

**Comment by jonashaag** [comment]
Oops I undid my Zstandard removal with the rebase. Will rebase and fix

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jonashaag

**Pull request merged**